keras/callbacks.py

```diff
@@ -45,7 +45,8 @@ def set_params(self, params):
 
     def set_model(self, model):
         for callback in self.callbacks:
-            callback.set_model(model)
+            if not hasattr(callback, 'model'):
+                callback.set_model(model)
```

Comment:
Prefer setting `self.model = None` in constructor then checking `if not self.model`

Reply:
Agreed, and updated.